mmdeploy/codebase/mmpose/deploy/pose_detection_model.py

```diff
@@ -159,6 +159,94 @@ def pack_result(self,
         return data_samples
 
 
+@__BACKEND_MODEL.register_module('yolox-pose_end2end')
+class YoloXPoseEnd2EndModel(End2EndModel):
+    """End to end model for inference of pose detection.
+
+    Args:
+        backend (Backend): The backend enum, specifying backend type.
+        backend_files (Sequence[str]): Paths to all required backend files(e.g.
+            '.onnx' for ONNX Runtime, '.param' and '.bin' for ncnn).
+        device (str): A string represents device type.
+        deploy_cfg (str | mmengine.Config): Deployment config file or loaded
+            Config object.
+        model_cfg (str | mmengine.Config): Model config file or loaded Config
+            object.
+        data_preprocessor (dict | nn.Module | None): Input data pre-
+                processor. Default is ``None``.
+    """
+
+    def __init__(self,
+                 backend: Backend,
+                 backend_files: Sequence[str],
+                 device: str,
+                 deploy_cfg: Union[str, mmengine.Config] = None,
+                 model_cfg: Union[str, mmengine.Config] = None,
+                 data_preprocessor: Optional[Union[dict, nn.Module]] = None,
+                 **kwargs):
+        super(End2EndModel, self).__init__(
+            deploy_cfg=deploy_cfg, data_preprocessor=data_preprocessor)
+        self.deploy_cfg = deploy_cfg
+        self.model_cfg = model_cfg
+        self.device = device
+        self._init_wrapper(
+            backend=backend,
+            backend_files=backend_files,
+            device=device,
+            **kwargs)
+
+    def forward(self,
+                inputs: torch.Tensor,
+                data_samples: List[BaseDataElement],
+                mode: str = 'predict',
+                **kwargs):
+        """Run forward inference.
+
+        Args:
+            inputs (torch.Tensor): Input image(s) in [N x C x H x W]
+                format.
+            data_samples (List[BaseDataElement]): A list of meta info for
+                image(s).
+
+        Returns:
+            list: A list contains predictions.
+        """
+        assert mode == 'predict', \
+            'Backend model only support mode==predict,' f' but get {mode}'
+        inputs = inputs.contiguous().to(self.device)
+        batch_outputs = self.wrapper({self.input_name: inputs})
+        batch_outputs = self.wrapper.output_to_list(batch_outputs)
+        batch_outputs = [batch_output.cpu() for batch_output in batch_outputs]
+        results = self.pack_result(batch_outputs, data_samples)
+        return results
+
+    def pack_result(self,
```

Comment:
it seems the init and forward are almost same, could remove these new class add a `pack_yolox_pose_result` method to the `End2EndModel` and call it in the forward